libs/sdk-ui-kit/src/RecurrenceForm/useCronValidation.ts

```diff
@@ -0,0 +1,146 @@
+            }
+
+            const isValidCron = parseCron(expression);
+            if (!isValidCron) {
+                return RecurrenceFormCronErrorTypes.WRONG_FORMAT;
+            }
+
+            return null;
+        },
+        [allowHourlyRecurrence, parseCron],
+    );
+
+    const handleValidation = useCallback(
+        (expression: string) => {
+            const validationResult = validate(expression);
+
+            if (validationResult) {
+                switch (validationResult) {
+                    case RecurrenceFormCronErrorTypes.EMPTY:
+                        setCronError(errorMessages.empty.id);
+                        break;
+                    case RecurrenceFormCronErrorTypes.WRONG_FORMAT:
+                        setCronError(errorMessages.wrongFormat.id);
+                        break;
+                    case RecurrenceFormCronErrorTypes.FREQUENCY_TOO_HIGH:
+                        setCronError(errorMessages.frequencyTooHigh.id);
+                        break;
+                }
+            } else {
+                setCronError(null);
+            }
+
+            return !validationResult;
+        },
+        [validate],
+    );
+
+    const handleChange = useCallback(
+        (expression: string) => {
+            const validationResult = validate(expression);
+
+            if (cronError) {
+                if (validationResult) {
+                    switch (validationResult) {
+                        case RecurrenceFormCronErrorTypes.EMPTY:
+                            setCronError(errorMessages.empty.id);
+                            break;
+                        case RecurrenceFormCronErrorTypes.WRONG_FORMAT:
+                            setCronError(errorMessages.wrongFormat.id);
+                            break;
+                        case RecurrenceFormCronErrorTypes.FREQUENCY_TOO_HIGH:
+                            setCronError(errorMessages.frequencyTooHigh.id);
+                            break;
+                    }
+                } else {
+                    setCronError(null);
+                }
+            }
+            onCronValueChange(expression, !validationResult);
+        },
+        [cronError, validate, onCronValueChange],
+    );
+
+    const handleOnBlur = useCallback(
+        (value: string) => {
+            const validationResult = validate(value);
+            // NESTOR
+            if (validationResult) {
+                switch (validationResult) {
+                    case RecurrenceFormCronErrorTypes.EMPTY:
+                        setCronError(errorMessages.empty.id);
+                        break;
+                    case RecurrenceFormCronErrorTypes.WRONG_FORMAT:
+                        setCronError(errorMessages.wrongFormat.id);
+                        break;
+                    case RecurrenceFormCronErrorTypes.FREQUENCY_TOO_HIGH:
+                        setCronError(errorMessages.frequencyTooHigh.id);
+                        break;
```

Comment:
"default" branch should be always defined. You leave unhandled validation result bubble away.

Reply:
Fixed.